### PR TITLE
[Merged by Bors] - feat(Counterexamples/GameMultiplication): pre-game product cannot be lifted to games

### PR DESCRIPTION
**Pull request by vihdzp**
We show the existence of pre-games `x₁, x₂, y` with `x₁ ≈ x₂` but `x₁ * y ≉ x₂ * y`.

---

- [x] depends on: #15766

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Game/Basic.lean [hunk 1]
```diff
@@ -412,6 +412,20 @@ instance isEmpty_rightMoves_mul (x y : PGame.{u})
...
 
+instance inhabited_leftMoves_mul (x y : PGame.{u})
+    [Inhabited (x.LeftMoves × y.LeftMoves ⊕ x.RightMoves × y.RightMoves)] :
+    Inhabited (x * y).LeftMoves := by
+  cases x
+  cases y
+  assumption
+
+instance inhabited_rightMoves_mul (x y : PGame.{u})
+    [Inhabited (x.LeftMoves × y.RightMoves ⊕ x.RightMoves × y.LeftMoves)] :
```
[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/SetTheory/Game/Basic.lean:423]: I started a Zulip discussion https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/Inhabited.20.28Sum.20_.20_.29

[the remark this answers is not in the document]
Comment (vihdzp) [reply Mathlib/SetTheory/Game/Basic.lean:423]: I can turn this into a `Nonempty` instance, I'm really only using it to get an arbitrary element.

[the remark this answers is not in the document]
Comment (vihdzp) [reply Mathlib/SetTheory/Game/Basic.lean]: Actually, on closer inspection, my element isn't really arbitrary. I've removed the `Inhabited` instance altogether.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**